common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Collection.scala

```diff
@@ -0,0 +1,10 @@
+package uk.ac.wellcome.models.work.internal
+
+case class Collection(
+  label: Option[String],
+  path: String,
+) {
+  val paths: List[String] =
+    path.split("/").inits.toList.reverse.drop(0).map(_.mkString("/"))
```

Comment:
:thinking: it seems to me like it would be more logical modelling `path` as a `List[String]` (which is serialised / deserialised as a single `/` separated string in Elasticsearch for purposes of display, search etc). But maybe not necessary due to the fact that it comes that way from Sierra, and we want it like that for the token analyser to do it's magic etc. What do other people think?

Reply:
> @warrd could you explain any benfits to making each path a List?
> Would you then expect paths to be List[List[String]]?

No about the `List[List[String]]`!!

What I mean is, instead of representing a path such as `"SA/MWF/J/1"` as a `String`, when in Scala land it would be a `List[String]`: i.e. `List("SA", "MWF", "J", "1")`.

The reason for doing this is that it is makes use of Scala's type system to more accurately depict what we are modelling, and removes unnecessary details (such as the fact that `/` is a seperator). Things like the depth of the path or what is the root come out very easily (i.e. `path.length` or `path.head`), and it is easy to manipulate the path independently of the encoding (i.e. can concatenate to the end without having to worry about what separator is used etc).

However, as I'm not sure we are going to be manipulating these paths very much, it might be unnecessary. I guess I tend to err on the side of using types to provide extra detail when possible, but a `String` might be fine in this case. Was just throwing it out there to see what others thought.

Reply:
It makes perfect sense - if we need to start doing this we can always change the model.

The model - as it's currently used for index information as well, will need to maintain the string with the separator and other information that the tokenizer uses. We should probably have that config central if that's how we want to do it.

I am erring on the side of, I will do that when we have the requirement to as it's a small, non-breaking change.

Reply:
> No about the `List[List[String]]`!!

I meant if
```
type Path = List[String]
type Paths = List[Path]

/*
where:
path = List("things", "to", "things")
paths = List(
  List("things"),
  List("things", "to"),
  List("things", "to", "things")
)
*/
```